src/UI/Art/Animation/BaseUnitAnimation.java

```diff
@@ -0,0 +1,85 @@
+package UI.Art.Animation;
+
+import java.awt.Graphics;
+import java.util.ArrayList;
+
+import Engine.XYCoord;
+import UI.Art.SpriteArtist.SpriteLibrary;
+import UI.Art.SpriteArtist.SpriteMapView;
+import UI.Art.SpriteArtist.UnitSpriteSet;
+import UI.Art.SpriteArtist.UnitSpriteSet.AnimState;
+import Units.Unit;
+
+public class BaseUnitAnimation implements GameAnimation
+{
+  protected long startTime = 0;
+  protected long duration = -42;
+
+  protected int tileSize;
+
+  protected Unit actor;
+  private UnitSpriteSet actorSpriteSet;
+  protected XYCoord actorCoord;
+
+  public BaseUnitAnimation(int tileSize, Unit actor, XYCoord actorCoord)
```

Comment:
This class isn't used for unit idling. Maybe it should be `BaseUnitActionAnimation` for clarity?